HardwareObjects/abstract/AbstractDiffractometer.py

```diff
@@ -0,0 +1,566 @@
+                if isinstance(motor, str, unicode):
+                    try:
+                        motors_positions_dict[str(motor)] = self.motors_hwobj[
+                            motor
+                        ].get_position()
+                    except KeyError:
+                        logging.getLogger("HWR").error("Invalid motor name (%s)", motor)
+                else:
+                    motors_positions_dict[motor.name] = float(motor.get_position())
+        else:
+            for role, motor in iter(self.motors_hwobj.items()):
+                motors_positions_dict[role] = float(motor.get_position())
+
+        return motors_positions_dict
+
+    def getPositions(self, motors_positions_list=[]):
+        warn(
+            "getPositions is deprecated. Use get_motor_positions instead",
+            DeprecationWarning,
+        )
+        return self.get_motor_positions(motors_positions_list)
+
+    def motor_positions_to_screen(self, centred_positions_dict):
+        """ Retirns x, y coordinates of the centred point, calculated
+            from positions of the centring motors
+        Args:
+            centred_positions_dict (dict): role:position dictionary
+        Returns:
+            (tuple): x, y [pixels]
+        """
+
+        if self.use_sample_centring:
+            self.update_zoom_calibration()
+            if None in (self.pixels_per_mm_x, self.pixels_per_mm_y):
+                return 0, 0
+            phi_angle = math.radians(
+                self.centring_phi.direction * self.centring_phi.getPosition()
+            )
+            sampx = self.centring_sampx.direction * (
+                centred_positions_dict["sampx"] - self.centring_sampx.getPosition()
+            )
+            sampy = self.centring_sampy.direction * (
+                centred_positions_dict["sampy"] - self.centring_sampy.getPosition()
+            )
+            phiy = self.centring_phiy.direction * (
+                centred_positions_dict["phiy"] - self.centring_phiy.getPosition()
+            )
+            phiz = self.centring_phiz.direction * (
+                centred_positions_dict["phiz"] - self.centring_phiz.getPosition()
+            )
+            rot_matrix = numpy.matrix(
+                [
+                    math.cos(phi_angle),
+                    -math.sin(phi_angle),
+                    math.sin(phi_angle),
+                    math.cos(phi_angle),
+                ]
+            )
+            rot_matrix.shape = (2, 2)
+            inv_rot_matrix = numpy.array(rot_matrix.I)
+            dx, dy = (
+                numpy.dot(numpy.array([sampx, sampy]), inv_rot_matrix)
+                * self.pixels_per_mm_x
+            )
+
+            x = (phiy * self.pixels_per_mm_x) + self.beam_position[0]
+            y = dy + (phiz * self.pixels_per_mm_y) + self.beam_position[1]
+
+            return x, y
+        else:
+            raise NotImplementedError
+
+    """ Head type and modes """
+
+    def get_head_type(self):
```

Comment:
I think this method is expected to be implemented for each Diffractometer class inheriting from this AbstractDiffractometer and match the enum class defined for the head types.

```suggestion
    def get_head_type(self):
        raise NotImplementedError
```
And then also remove the **try/except** block as @IvarsKarpics commented in line 199.